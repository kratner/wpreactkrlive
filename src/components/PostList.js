import React, { Component } from 'react';
import { appGraphQLUrl, gqlPosts } from './GetDataGraphQL';

class PostList extends Component {
  state = {
    posts: []
  };

  componentDidMount() {
    let appUrl = `${appGraphQLUrl}`;
    let queryPosts = `${gqlPosts}`;
    fetch(appUrl, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: queryPosts
    })
      .then(data => data.json())
      .then(data => {
        this.setState({
          posts: data.data.posts.edges
        });
      });
  }

  render() {
    let listposts = this.state.posts.map((node, index) => {
      return (
        <div key={index}>
          <h4>{node.node.title}</h4>
          <p
            dangerouslySetInnerHTML={{
              __html: node.node.content
            }}
          />
          {/* dangerouslySetInnerHTML eliminates the html tags */}
        </div>
      );
    });

    return (
      <div>
        <h1>Posts</h1>
        {listposts}
      </div>
    );
  }
}

export default PostList;
