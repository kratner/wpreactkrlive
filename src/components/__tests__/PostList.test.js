import React from "react";
import { shallow } from "enzyme";
import PostList from "../PostList";

describe("PostList", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<PostList />);
    expect(wrapper).toMatchSnapshot();
  });
});
