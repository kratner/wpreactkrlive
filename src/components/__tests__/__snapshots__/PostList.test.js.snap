// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList renders correctly 1`] = `ShallowWrapper {}`;
